sklearn/cross_decomposition/pls_.py

```diff
@@ -31,7 +31,11 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
     similar to the Power method for determining the eigenvectors and
     eigenvalues of a X'Y.
     """
-    y_score = Y[:, [0]]
```

Comment:
This does not produce a column containing only zeros. It slices the first column from Y

Reply:
Yes, you are right. However, when the first column of Y consists of only zeros, y_score gets initialized as a column containing only zeros. I intend to avoid this here.